File: eau04.py
import sys

# fonctions utilisées
def nb_premier(nb):
	i = 2
	if nb <= 1:
		return -1
	else:
		while i < nb:
			if nb % i == 0:
				return 0
			i += 1
	return 1

def find_next_nb_premier(nb):
	if nb_premier(nb) == 1:
		return nb
	nb += 1
	while nb_premier(nb) == 0:
		nb += 1
	return nb

def is_args_valid():
	if is_nb_arg_correct == False:
		return False
	if sys.argv[1].isdigit() == False:
		return False


# Partie 1 : Gestion d'erreur
is_nb_arg_correct = True if len(sys.argv) == 2 else False

if is_args_valid() == False:
	print(-1)
	exit()

# Partie 2 : Parsing
nombre = int(sys.argv[1])


# Partie 3 : Résolution
next_nb_premier = find_next_nb_premier(nombre)

# Partie 4 : Affichage
print(next_nb_premier)

File: eau04.js
// fonctions utilisées

function nb_premier(nb)
{
	let i = 2;
		while (i < nb)
		{
			if (nb % i === 0)
				return 0;
			i++;
		}
	return 1;
}

function find_next_nb_premier(nb)
{
	if (nb_premier(nb) === 1)
		return nb;
	nb++;
	while (nb_premier(nb) === 0)
		nb++;
	return nb;
}

function is_arg_valid()
{
	if (is_nb_arg_correct === false)
		return false;
	if (isNaN(process.argv[2]) == true)
		return false;
	if (process.argv[2][0] === '-')
		return false;
}

// Partie 1 : Gestion d'erreur
is_nb_arg_correct = process.argv.length === 3;

if (is_arg_valid() === false)
{
	console.log("-1");
	process.exit(1);
}

// Partie 2 : Parsing
const number = parseInt(process.argv[2])

// Partie 3 : Résolution
const next_nb_premier = find_next_nb_premier(number);

// Partie 4 : Affichage
console.log(next_nb_premier);